great_expectations/core/expectation_suite.py

```diff
@@ -219,12 +220,13 @@ def _process_expectation(
     def delete_expectation(self, expectation: Expectation) -> Expectation:
         """Delete an Expectation from the collection.
 
+        Example:
+            >>> suite.delete_expectation(suite.expectations[0])
+
         Raises:
             KeyError: Expectation not found in suite.
         """
-        remaining_expectations = [
-            exp for exp in self.expectations if exp.configuration != expectation.configuration
-        ]
+        remaining_expectations = [exp for exp in self.expectations if exp != expectation]
```

Comment:
Should this be using the same kind of logic as in `_determine_if_expectation_is_unique` wrt ignoring ids?